File: log_analyzer/tests/fixtures/NotInLoop.java
package com.example.chatty;

import android.util.Log;
import java.util.List;
import java.util.Optional;

public class NotInLoop {
    void afterLoop(List<String> items) {
        Log.i("T", "before");
        for (String item : items) {
            Log.d("T", "inside for");
            consume(item);
        }
        Log.d("T", "after for");
        items.forEach(this::consume);
        Log.i("T", "after forEach");
        items.stream().map(String::trim).forEach(this::consume);
        Log.w("T", "after stream");
        Optional.ofNullable("x").map(v -> v + "y");
        Log.e("T", "after optional");
    }

    void consume(String s) {}
}

File: log_analyzer/tests/fixtures/NotInLoop.kt
package com.example.chatty

import android.util.Log

class NotInLoop {
    fun afterLoop(items: List<String>) {
        Log.i("T", "before")
        for (item in items) {
            Log.d("T", "inside for")
        }
        Log.d("T", "after for")
        items.forEach { consume(it) }
        Log.i("T", "after forEach")
        val names = items.map { it.length }
        Log.w("T", "after map $names")
        repeat(3) { consume("x") }
        Log.e("T", "after repeat")
    }

    fun consume(s: String) {}

    fun usesMap(items: List<String>) {
        Log.d("T", items.map { it }.toString())
    }
}
